streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1426,6 +1425,9 @@ private void verifyMaxInFlightRequestPerConnection(final Object maxInFlightReque
     public Serde defaultKeySerde() {
         final Object keySerdeConfigSetting = get(DEFAULT_KEY_SERDE_CLASS_CONFIG);
         try {
+            if (keySerdeConfigSetting ==  null) {
+                return null;
```

Comment:
It's been a while, but I am still wondering if we should throw a `ConfigException` directly instead of returning `null`? And this would be the only place in the code when we throw a `ConfigException` for this case. Below, there is some repetitive code that calls `defaultKeySerde` (or `defaultValueSerd`) and throws `ConfigException` if `null` is returned.

Reply:
I don't think we want to, this is called when we create the `AbstractProcessorContext` which happens before we actually set the serdes. Since we don't check the configs and actually set the serde till later, I don't think we want to throw an exception here

Reply:
Well, I think to argue that we might want to change the `AbstractProcessorContext`? Instead of making the call in the constructor, we update `keySerde()` and `valueSerde()` method to make the call:
```
public Serde<?> keySerde() {
    if (keySerde == null) {
        keySerde = config.defaultValueSerde();
    }
    return keySerde;
}
```

Thoughts?